### PR TITLE
Rule S2077: support of more database libraries (#3293)

### LINKED ISSUE #3293: Rule S2077: support of more database libraries
Some database librairies like Mysql.Data are not supported:
```
MySqlConnection myConnection = new MySqlConnection(strConn);
string sensitiveString2 = "SELECT * FROM Table1 WHERE col1 = '" + param + "'";
MySql.Data.MySqlClient.MySqlHelper.ExecuteNonQuery(myConnection, sensitiveString2); // FN
```
Ideally S2077 should follow rule S3649 (all the sinks defined in S3649 should be supported in S2077)

Related discussions:
- https://github.com/SonarSource/sonar-dotnet/issues/2800#issuecomment-584036701
- https://jira.sonarsource.com/browse/RSPEC-4873?focusedCommentId=232648&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-232648

Eric

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**
Fixes #3293

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/ExecutingSqlQueriesBase.cs [hunk 1]
```diff
@@ -41,13 +43,63 @@ public abstract class ExecutingSqlQueriesBase<TSyntaxKind, TExpressionSyntax> :
...
+            };
+
+        private readonly MemberDescriptor[] properties =
+            {
+                new MemberDescriptor(KnownType.System_Data_Odbc_OdbcCommand, "CommandText"),
+                new MemberDescriptor(KnownType.System_Data_OracleClient_OracleCommand, "CommandText"),
+                new MemberDescriptor(KnownType.System_Data_SqlClient_SqlCommand, "CommandText"),
+                new MemberDescriptor(KnownType.System_Data_SqlServerCe_SqlCeCommand, "CommandText"),
+                new MemberDescriptor(KnownType.MySql_Data_MySqlClient_MySqlCommand, "CommandText"),
+                new MemberDescriptor(KnownType.Microsoft_Data_Sqlite_SqliteCommand, "CommandText"),
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/ExecutingSqlQueriesBase.cs:95]: I think there's and issue with `System.Data.Sqllite` and that is, that there exists
`System.Data.Sqlite` for .NET Fx and .NET Core 2.1
`System.Data.SQLite` for .NET Core 3.1 and .NET 5
It's documented in sonar-security ITs and I'm not sure if current configuration supports both (probably not yet).

Can you take a look and if it's confirmed, then add tests for netcore with `SQLite`?

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
I left few minor comments and one about adding more for SQLite

File: analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/ExecutingSqlQueriesBase.cs (hunk 1 above)
Comment (csaba-sagi-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/ExecutingSqlQueriesBase.cs:95]: I could not find the nuget package version with System.Data.Sqlite.

Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/ExecutingSqlQueriesBase.cs:95]: That was probably related to the `Microsoft.Data.Sqlite` => `System.Data.SQLite` move

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ExecutingSqlQueries_Net46.cs [hunk 2]
```diff
@@ -106,7 +110,7 @@ public void OdbcCommands(OdbcConnection connection, OdbcTransaction transaction,
             OdbcDataAdapter adapter;
             adapter = new OdbcDataAdapter(); // Compliant
             adapter = new OdbcDataAdapter(command); // Compliant
-            adapter = new OdbcDataAdapter(query, ""); // Compliant
+            adapter = new OdbcDataAdapter(query, $" non constant {query}"); // Compliant
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ExecutingSqlQueries_Net46.cs:113]: This is little confusing, as it looks that the type has two query parameters and it's not clear why we don't raise on the second one.
```suggestion
            adapter = new OdbcDataAdapter(query, $"concatenated connection string {query}"); // Compliant
```
or with new parameter in the method
```suggestion
            adapter = new OdbcDataAdapter(query, $"connection string {server}"); // Compliant
```

Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ExecutingSqlQueries_Net46.cs:113]: Applies to all

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM, please take a look at the last comment before merging

**Pull request merged**